scripts/ci/test_reborn_pr_test_plan.py

```
@@ -191,6 +191,48 @@ def test_recorded_fixture_change_runs_only_qa_replay(self) -> None:
+        self.assertEqual(plan["mode"], "none")
+        self.assertEqual(plan["crate_buckets"], [])
+        self.assertEqual(plan["integration_lanes"], [])
+        self.assertEqual(plan["root_partitions"], [])
+        self.assertTrue(plan["run_qa_replay"])
+        self.assertTrue(
+            any("Reborn E2E workflow owns" in reason for reason in plan["reasons"]),
+            plan["reasons"],
+        )
+
+    def test_shared_e2e_harness_remains_an_explicit_mapping_error(self) -> None:
+        with self.assertRaisesRegex(ValueError, "unmapped Reborn test path"):
```

Ruff (0.16.0)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)